### ISSUE TITLE
Externalizing DB password is not working

### ISSUE THREAD

**Issue by DwijadasDey** [confluentinc/kafka-connect-jdbc#735]
Hi
I am trying to externalizing user name and password for oracle DB using `FileConfigProvider` provider. 

For that i have created a properties file that contains user name and password.

```
$ cat /home/kfk/data/ora_credentials.properties
ora.username="apps"
ora.password="Passw0rd!"
```

Added the config providers as file and also the config.providers.file.class as FileConfigProvider in the worker config:

```
$ cat /home/kfk/etc/kafka/connect-distributed.properties
...
...
config.providers=file
config.providers.file.class=org.apache.kafka.common.config.provider.FileConfigProvider
...
...
```

Restarted the worker and submitted a task using REST with the following config

```
"config": {
               "connector.class": "io.confluent.connect.jdbc.JdbcSourceConnector",
               "tasks.max": "1",
	       "connection.user": "${file:/home/kfk/data/ora_credentials.properties:ora.username}",
               "connection.password": "${file:/home/kfk/data/ora_credentials.properties:ora.password}",
               ...
               ...
}
```
Submitting the above task resulting in the following error:

```
{
  "error_code": 400,
  "message": "Connector configuration is invalid and contains the following 2 error(s):\nInvalid value java.sql.SQLException: ORA-01017: invalid username/password; logon denied\n for configuration Couldn't open connection to jdbc:oracle:thin:@oebsr122.infodetics.com:1521:VIS\nInvalid value java.sql.SQLException: ORA-01017: invalid username/password; logon denied\n for configuration Couldn't open connection to jdbc:oracle:thin:@oebsr122.infodetics.com:1521:VIS\nYou can also find the above list of errors at the endpoint `/{connectorType}/config/validate`"
}
```

Assuming the above config does not replaces the user name and password at all rather entire values for connection.user and connection.password are  used to connect to the DB resulting in ORA-01017: invalid username/password error.

Where could be the issue ?

**Comment by DwijadasDey** [comment]
In the distributed connector log, i can see value of username is correctly replaced from the properties file and the password is hidden. But still there is an error code of 400 about invalid values ?

![Screenshot_1](https://user-images.githubusercontent.com/12824049/67956685-860f2700-fc1a-11e9-889a-b5c0544a5b8c.png)

**Comment by adityajalkhare** [comment]
Hi @DwijadasDey , I am facing this same issue currently.

my `credentials.properties` file -
```
source-url=jdbc:sqlserver://testdb1.database.windows.net:1433;databaseName=db1234
source-username=testuser
source-password=P#5sW0rd
```
Full description here: [jdbc-source-connector-not-working-with-fileconfigprovider-envconfigprovider](https://stackoverflow.com/questions/63281321/jdbc-source-connector-not-working-with-fileconfigprovider-envconfigprovider)

Any help is appreciated!!

**Comment by DwijadasDey** [comment]
Hi
You can try out the following diagnostic steps to pinpoint the actual cause:

**1.**  Double check if you have added the following lines in the connect distributed properties file:

...
...
```
config.providers=file
config.providers.file.class=org.apache.kafka.common.config.provider.FileConfigProvider
```
...
...

It looks like you have already added these two lines, since sink connector is working  

**2.** From the error message `Login failed for user '${file:/opt/confluentinc-kafka-connect-jdbc/etc/connect-secrets.properties:source-username}'` it looks like actual username is not picked up from the properties file. 

I guess you probably have some extra characters like spaces at the end of lines in the properties file(**connect-secrets.properties**) or check if the path `/opt/confluentinc-kafka-connect-jdbc/etc/connect-secrets.properties` and the identifier `source-username` is correct or not.

For debugging, run the connect distributed process like following and thereby enabling you to view the parameters picked up from the properties file in the console.

`# connect-distributed /path/to/connect-distributed.properties  & echo $! > /path/to/connect-distributed.pid &`

Regards

**Comment by adityajalkhare** [comment]
Hi @DwijadasDey , Thanks for your help!!!
We found it to be the issue with our base image. We had been using `bitnami/kafka` image extended with custom plugins and some plugins from `confluentinc/cp-kafka-connect` image. Sink had no issue in this, but in Source, connector tried to login to db even before letting `FileConfigProvider` resolve the values.

We switched back to `confluentinc/cp-kafka-connect-base` and extended it with required plugins. Now this config works.
